src/os/freertos/etcpal/os_queue.c

```diff
@@ -115,7 +120,38 @@ bool etcpal_queue_is_empty(const etcpal_queue_t* id)
 
 bool etcpal_queue_is_empty_from_isr(const etcpal_queue_t* id)
 {
-  UBaseType_t numMessages = uxQueueMessagesWaitingFromISR(*id);
+  // UBaseType_t numMessages = uxQueueMessagesWaitingFromISR(*id);
+
+  // return (numMessages == 0);
+  return xQueueIsQueueEmptyFromISR(*id);
+}
+
+bool etcpal_queue_is_full(const etcpal_queue_t* id)
+{
+  UBaseType_t numMessages = uxQueueSpacesAvailable(*id);
 
   return (numMessages == 0);
 }
+
+bool etcpal_queue_is_full_from_isr(const etcpal_queue_t* id)
+{
+  return xQueueIsQueueFullFromISR(*id);
+}
+
+unsigned int etcpal_queue_size(const etcpal_queue_t* id)
+{
+  UBaseType_t numMessages = uxQueueMessagesWaiting(*id);

```

Comment:
For these functions, why the extra local var? Why not just

```c
return uxQueueMessagesWaiting(*id);
```

Reply:
Copy paste to keep similarity with previous code implementation. I can change this around the whole file tho

Reply:
Yeah, let's do if you don't mind.